Sources/Extensions/Alamofire/SessionManager+Extensions.swift

```diff
@@ -26,18 +26,64 @@ import RxAlamofire
 
 public extension Reactive where Base: SessionManager {
 
+    /**
+     Creates an observable of the `Request`.
+     
+     - parameter method: Alamofire method object
+     - parameter url: An object adopting `URLConvertible`
+     - parameter parameters: An array of dictionaries containing all necessary options
+     - parameter encoding: The kind of encoding used to process parameters
+     - parameter header: A dictionary containing all the additional headers
+     
+     - returns: An observable of the `Request`
+     */
+    public func request(_ method: Alamofire.HTTPMethod,
+                        _ url: URLConvertible,
+                        parameters: [Parameters]? = nil,
+                        encoding: ParameterEncoding = URLEncoding.default,
+                        headers: [String: String]? = nil
+        )
+        -> Observable<DataRequest>
+    {
+        return Observable.deferred {
+            var urlRequest = URLRequest(url: try url.asURL())
+            urlRequest.httpMethod = method.rawValue
+            urlRequest.allHTTPHeaderFields = headers
+            urlRequest.httpBody = try? JSONSerialization.data(withJSONObject: parameters, options: [])
+            
+            if urlRequest.value(forHTTPHeaderField: "Content-Type") == nil {
```

Comment:
looks like this check is not required since you just create new URLRequest, so it should has no headers (but check yourself).

Reply:
you should use this method https://github.com/Alamofire/Alamofire/blob/3e488dcd5caf8d76f3da79c7b6d7f4d784ca3296/Source/ParameterEncoding.swift#L40 of `encoding` argument

Reply:
and this one https://github.com/Alamofire/Alamofire/blob/3e488dcd5caf8d76f3da79c7b6d7f4d784ca3296/Source/ParameterEncoding.swift#L319 for array.
But there still a problem with `URLEncoding` and array of parameters...